rusty_socket_server/src/socket_server.rs

```diff
@@ -1,26 +1,33 @@
 use std::io::{Read, Write};
-use std::net::{SocketAddrV4, TcpListener, TcpStream};
+use std::net::{SocketAddr, SocketAddrV4, TcpListener, TcpStream, ToSocketAddrs};
+use std::thread;
+use std::sync::{Mutex, Arc};
 
-use crate::Connections;
-use rusty_socket_core::{DataFrame, HandShake, OpCode};
+use crate::{Connection, HandShake, SsError};
+use crate::Result;
 
 pub struct SocketServer {
     target: SocketAddrV4,
-    connections: Vec<Connections>,
+    active_connections: Arc<Mutex<Vec<TcpStream>>>,
 }
 
 impl SocketServer {
-    pub fn new() -> Self {
-        SocketServer {
-            target: "127.0.0.1:8080".parse::<SocketAddrV4>().unwrap(),
-            connections: Vec::new(),
+    pub fn build(address: impl ToSocketAddrs) -> Result<Self> {
```

Comment:
Usually server would be listening on localhost and specified port. Just a suggestion based on my assumption.

Reply:
Yes this will be changed